### ISSUE TITLE
Azure invalid content length errors being reported as content-type error 

### ISSUE THREAD

**Issue by amarzavery** [Azure/azure-storage-node#202]
Kind of ran into this sideways.

I was accidentally setting an invalid content length value as a string that couldn't be parsed to a number. The error from the sdk is:

```
{"msec":204.5356139987707,"error":"StorageError: Content-Type response header is missing or invalid.","data":{"data":null,"isBoom":true,"isServer":false,"output":{"statusCode":400,"payload":{"statusCode":400,"error":"Bad Request","message":"StorageError: Content-Type response header is missing or invalid."},"headers":{}}}}
```

However, the error being returned from Azure is:

```
{ isSuccessful: false,
statusCode: 400,
body: '<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01//EN""http://www.w3.org/TR/html4/strict.dtd">\r\n<HTML><HEAD><TITLE>Bad Request</TITLE>\r\n<META HTTP-EQUIV="Content-Type" Content="text/html; charset=us-ascii"></HEAD>\r\n<BODY><h2>Bad Request - Invalid Content Length</h2>\r\n<hr><p>HTTP Error 400. There is an invalid content length or chunk length in the request.</p>\r\n</BODY></HTML>\r\n',
headers: 
 { 'content-type': 'text/html; charset=us-ascii',
   server: 'Microsoft-HTTPAPI/2.0',
   date: 'Wed, 28 Sep 2016 01:44:48 GMT',
   connection: 'close',
   'content-length': '374' },
md5: undefined }
```

This is the stack trace that was reported:

```
StorageError: Content-Type response header is missing or invalid.
  at Function.StorageServiceClient._normalizeError (/usr/local/apps/makersteps/node_modules/azure-storage/lib/common/services/storageserviceclient.js:1146:23)
  at BlobService.StorageServiceClient._processResponse (/usr/local/apps/makersteps/node_modules/azure-storage/lib/common/services/storageserviceclient.js:706:50)
  at Request.processResponseCallback [as _callback] (/usr/local/apps/makersteps/node_modules/azure-storage/lib/common/services/storageserviceclient.js:306:37)
  at Request.self.callback (/usr/local/apps/makersteps/node_modules/request/request.js:187:22)
  at emitTwo (events.js:106:13)
  at Request.emit (events.js:191:7)
  at Request.<anonymous> (/usr/local/apps/makersteps/node_modules/request/request.js:1044:10)
  at emitOne (events.js:96:13)
  at Request.emit (events.js:188:7)
  at IncomingMessage.<anonymous> (/usr/local/apps/makersteps/node_modules/request/request.js:965:12)

```

Using azure-storage 1.3.0 and hapi 15.0.1.

Actual issue was reported in the node sdk https://github.com/Azure/azure-sdk-for-node/issues/1908 by @geuis

**Comment by hasonmsft** [comment]
@geuis, thanks for reporting this. Will look into this and get you updated. 
At the same time, could you share the which SDK (package name & the version) you are using?

**Comment by geuis** [comment]
@hasonmsft Thanks for the update. I'm currently using azure-storage 1.3.0.

**Comment by hasonmsft** [comment]
@geuis , the current error handling can only handle the normal server error case - `xml` format error payload from the server. But as you sent incorrect format request (non-numeric content-length) and the request failed before the real server handling logic and returned `html` format error. We'll treat this as abnormal response and return `StorageError: Content-Type response header is missing or invalid.` To support HTML format error response, it needs more effort. 

For your case, you can get the detailed information from the response when getting this kind of error as the workaround. 

Will put this into our backlog and get you updated when we decide to fix it. Please let me know if this works for you or not.

**Comment by hasonmsft** [comment]
@geuis , this has been fixed in the 1.4.0 release.
